### DISCUSSION TITLE
did git reset hard but my remote is ahead in commits

### DISCUSSION THREAD

**Discussion by dmastrop** [community/community#63665] [Programming Help] [answered]
### Body

hi All

I had to do a git reset hard to get back to a commit that I wanted and that worked locally. 

I forgot that this would cause an issue with the remote repository when I try to push. The remote is a few commits ahead.

How can I get I get the push of this local repository (on the commit that I want) to the remote repository.

I am also using Jenkins  which might complicate things.

Should I pull the current remote repository back and try to work from there?(as suggested)

here is the error message that I am getting right now when trying to push to the remote repo after the git reset hard.

git push origin master
To https://github.com/dmastrop/Terraform-Cloud9-AWS_Prometheus_Grafana_Jenkins_ansible.git
 ! [rejected]        master -> master (non-fast-forward)
error: failed to push some refs to 'https://github.com/dmastrop/Terraform-Cloud9-AWS_Prometheus_Grafana_Jenkins_ansible.git'
**hint: Updates were rejected because the tip of your current branch is behind**
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

warm regards

Dave

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by nirav-gajera** [comment, marked as answer]
you should pull the current remote repository back and try to work from there. This will bring your local repository up to date with the remote repository, and you will be able to push your changes without any problems.

To do this, you can use the following command:

```
git pull origin master
```

This command will fetch the latest changes from the remote repository and merge them into your local repository. If there are any conflicts, you will need to resolve them manually.

Once you have pulled the remote repository, you should be able to push your changes without any problems.

Here are the steps on how to pull the current remote repository back and try to work from there:

1. Run the following command to pull the remote repository:

```
git pull origin master
```

2. If there are any conflicts, resolve them manually.
3. Once you have resolved all of the conflicts, try to push your changes again.

**Comment by dmastrop** [comment]
hi Nirav thanks for your answer. I did this before you had replied. I think this is good to go. Can you confirm?

**Reply by nirav-gajera** [reply]
sure, first backup your folder sometime there's chance to lose some files
